### ISSUE TITLE
Multiple Profiles

### ISSUE THREAD

**Issue by Nick600-alt** [TeamNewPipe/NewPipe#5654]
### Checklist

- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### Describe the feature you want

Add the ability to make different profiles. Have one main profile and then smaller ones where you could subscribe to different channels. For example a main profile where all your subscriptions appear and then a smaller profile with gaming channels only, another for comedy only, etc.

FreeTube the desktop app has this feature and it makes it easy to organize different types of videos.

#### Additional context

Could be like a drop down menu in the top corner and you choose your profile or something. 

#### How will you/everyone benefit from this feature?

This helps users organize different types of videos, rather than having all videos in one spot.

**Comment by SameenAhnaf** [comment]
Your suggestions are really nice. But I think profile features can be far more extended.

Multiple profiles should act like multiple YouTube accounts. They should be directly accessed from an icon on the top right corner of any main page (As expected, different people manage tabs on main page in different ways. So, profile icon should appear in every single tab on main page) They should have seperate managements for
1. Subscriptions
2. Bookmarked playlists
3. Downloads
4. History
5. What's new
6. Settings

An "All in one" profile should be a good addition too. Only, its settings will be different from others. Its settings should be assigned as default settings for new profiles.

**Comment by HTSamurai** [comment]
This would be great. I've always wanted this but never got round to
creating an issue.

**Comment by demonff1** [comment]
My post #5093 elaborates on this. I suggested using subscription playlists to do this.
 Also if the what's new tab on newpipe doesn't show updates of all subs but a certain selected subscription playlist will also be helpful. I think my way will help achieve the goal without significant changes for the time being.

**Comment by SameenAhnaf** [comment]
> My post #5093 elaborates on this. I suggested using subscription playlists to do this.
> Also if the what's new tab on newpipe doesn't show updates of all subs but a certain selected subscription playlist will also be helpful. I think my way will help achieve the goal without significant changes for the time being.

That's good. But you only talked about "What's New" tab. My comment asks to keep seperate every single thing.

**Comment by SameenAhnaf** [comment]
> Your suggestions are really nice. But I think profile features can be far more extended.
> 
> Multiple profiles should act like multiple YouTube accounts. They should be directly accessed from an icon on the top right corner of any main page (As expected, different people manage tabs on main page in different ways. So, profile icon should appear in every single tab on main page) They should have seperate managements for
> 
> 1. Subscriptions
> 2. Bookmarked playlists
> 3. Downloads
> 4. History
> 5. What's new
> 6. Settings
> 7. Player info (Brightness, Volume, Speed, Pitch, Caption on/off, Fit/Fill/Zoom)
> 
> An "All in one" profile should be a good addition too. Only, its settings will be different from others. Its settings should be assigned as default settings for new profiles.

One more thing, there are support for some websites. Each site should not have multiple profiles. Rather, one profile should be able to access all sites. We might find similar channels on more than one site. That's why, each profile should be independent, not a feature under any website.

**Comment by Flam3z** [comment]
+1 for this, i'd like to also have a profile specifically for soundcloud or other services.

**Comment by Timothy099qw** [comment]
Hi, can I work on this feature if it is not closed? didn't see this feature available to enable on the app or other branches. Looking forward to any replies. Thanks.

**Comment by Timothy099qw** [comment]
> Newpipe is currently on a feature freeze pending a major rewrite. So I'm assigning this to you, but please wait for a dev to give you the go ahead before starting work on it, just in case this gets postponed to after the rewrite.

Ok, I will try some modifications see if they work atm, when dev notifies me I will start work on the updated version.

**Comment by Timothy099qw** [comment]
hi all, just opened a pull request for reviewing the idea of putting accounts (profiles) entry in settings. And users can login using the existing 2 accounts. If this implementation's idea is approved (Not necessarily PR) then I will continue on the code and wait until dev finishes refactoring, and make playlists to store a copy to res/data for separating different user accounts/profiles.

**Comment by offbeat-stuff** [comment]
I would like if multiple profiles can run multiple streams simultaneously. Not exactly play them at same time. but be able to switch from one to another without losing any information. Like fast switch between music and a news video. I currently use multiple newpipe apks (nightly, other forks) for this.

**Comment by Timothy099qw** [comment]
> I would like if multiple profiles can run multiple streams simultaneously. Not exactly play them at same time. but be able to switch from one to another without losing any information. Like fast switch between music and a news video. I currently use multiple newpipe apks (nightly, other forks) for this.

does ur device support multi-boxing?

**Comment by offbeat-stuff** [comment]
well yes, though that is more complicated than using multiple apps

**Comment by Souren-Araya** [comment]
Multiples profiles would be very useful for users speaking/learning multiple languages - each profile has it's own history, subscription, regional settings, attached google account, etc

It's not just about playlist/subscription grouping. It's also about having a whole set of different environments each of them having recommendations for the specific country in the specific language with the specific language settings for audio/subs, specific google account and so on.

**Comment by Vivekanandhan-Muthurasan** [comment]
Is anyone currently working on this feature?
